packages/celotool/src/lib/oracle.ts

```diff
@@ -17,25 +23,124 @@ export async function installHelmChart(celoEnv: string) {
     `--set image.tag=${fetchEnv(envVar.ORACLE_DOCKER_IMAGE_TAG)}`,
+    `--set kube.authTokenName=${kubeAuthTokenName}`,
+    `--set oracle.replicas=${replicas}`,
     `--set oracle.web3ProviderUrl=${getFornoUrl(celoEnv)}`,
-    `--set oracle.privateKeys=\\{${oraclePrivateKeys.join(',')}\\}`,
-  ]
+  ].concat(await oracleIdentityHelmParameters(celoEnv))
+}
+
+async function oracleIdentityHelmParameters(celoEnv: string) {
+  const addressesAndVaults = oracleAddressesAndVaults()
+  const replicas = addressesAndVaults.length
+  let params: string[] = []
+  for (let i = 0; i < replicas; i++) {
+    const identity = await createOracleIdentityIfNotExists(celoEnv, i)
+    const { address, keyVaultName } = addressesAndVaults[i]
+    const prefix = `--set oracle.identities[${i}]`
+    params = params.concat([
+      `${prefix}.address=${address}`,
+      `${prefix}.azure.id=${identity.id}`,
+      `${prefix}.azure.clientId=${identity.clientId}`,
+      `${prefix}.azure.keyVaultName=${keyVaultName}`,
+    ])
+  }
+  return params
 }
 
 function releaseName(celoEnv: string) {
   return `${celoEnv}-oracle`
 }
+
+async function createOracleIdentityIfNotExists(celoEnv: string, index: number) {
+  const identity = await createIdentityIfNotExists(oracleIdentityName(celoEnv, index))
+
+  // Grant the service principal permission to manage the oracle identity.
+  // See: https://github.com/Azure/aad-pod-identity#6-set-permissions-for-mic
+  const [rawServicePrincipalClientId] = await execCmdWithExitOnFailure(
+    `az aks show -n ${clusterName()} --query servicePrincipalProfile.clientId -g ${resourceGroup()} -o tsv`
+  )
+  const servicePrincipalClientId = rawServicePrincipalClientId.trim()
+  await execCmdWithExitOnFailure(
+    `az role assignment create --role "Managed Identity Operator" --assignee ${servicePrincipalClientId} --scope ${identity.id}`
+  )
+  // Allow the oracle identity to access the correct key vault
+  await execCmdWithExitOnFailure(
+    `az keyvault set-policy --name ${keyVaultName(
+      index
+    )} --key-permissions {get,list,sign} --object-id ${identity.principalId} -g ${resourceGroup()}`
+  )
+  return identity
+}
+
+function oracleIdentityName(celoEnv: string, index: number) {
+  return `${celoEnv}-oracle-${index}`
+}
+
+function keyVaultName(oracleIndex: number) {
+  return oracleAddressesAndVaults()[oracleIndex].keyVaultName
+}
+
+// Decodes the env variable ORACLE_ADDRESS_KEY_VAULTS of the form:
+//   <address>:<keyVaultName>,<address>:<keyVaultName>
+//   eg: 0x0000000000000000000000000000000000000000:keyVault0,0x0000000000000000000000000000000000000001:keyVault1
+// into an array in the same order
+function oracleAddressesAndVaults(): {
+  address: string
+  keyVaultName: string
+}[] {
+  const vaultNamesAndAddresses = fetchEnv(envVar.ORACLE_ADDRESS_KEY_VAULTS).split(',')
+  const addressesAndVaults = []
+  for (const nameAndAddress of vaultNamesAndAddresses) {
+    const [address, keyVaultName] = nameAndAddress.split(':')
```

Comment:
We should add a check here and fail if input isn't as expected (ex. address missing)